<!-- usage: roslaunch ur5lego spawn_block.launch block_name:=block1 block_type:=X1-Y1-Z2 -->

<launch>
    <arg name="block_type" />
    <arg name="block_name" />
    <arg name="spawn_x" default="0.2" />
	<arg name="spawn_y" default="0.5" />
	<arg name="spawn_z" default="1" />

    <!-- sends block XML to parameter server -->
    <param name="block_description" 
        command="$(find xacro)/xacro '$(find ur5lego)/blocks_description/$(arg block_type).urdf.xacro' block_name:=$(arg block_name)"/>

    <!-- spawn the model -->
    <!-- the node that spawns the model has the name of the model as part of its name
        to avoid having multiple nodes with the same name, which ros doesn't allow-->
    <node name="spawn_gazebo_model_$(arg block_name)" pkg="gazebo_ros" type="spawn_model"
        args="-urdf -param block_description -model $(arg block_name) -x $(arg spawn_x) -y $(arg spawn_y) -z $(arg spawn_z)"/>
</launch>